bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,483 @@
+        channelConfigsExt
+                .add(new SecondGenerationChannelConfig("ownConsumptionDay", "td", 100, SmartHomeUnits.KILOWATT_HOUR));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("ownConsRateDay", "td", 103, SmartHomeUnits.PERCENT));
+        channelConfigsExt
+                .add(new SecondGenerationChannelConfig("autonomyDegreeDay", "td", 106, SmartHomeUnits.PERCENT));
+        channelConfigsExt.add(
+                new SecondGenerationChannelConfig("homeConsumptionTotal", "td", 109, SmartHomeUnits.KILOWATT_HOUR));
+        channelConfigsExt
+                .add(new SecondGenerationChannelConfig("ownConsumptionTotal", "td", 112, SmartHomeUnits.KILOWATT_HOUR));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("totalOperatingTime", "td", 115, SmartHomeUnits.HOUR));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("current", "td", 118, SmartHomeUnits.AMPERE));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("currentDir", "td", 121, SmartHomeUnits.AMPERE));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("chargeCycles", "td", 124, null));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("batteryTemperature", "td", 127, SIUnits.CELSIUS));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("loginterval", "td", 130, SmartHomeUnits.MINUTE));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("s0InPulseCnt", "td", 133, null));
+        channelConfigsExt.add(new SecondGenerationChannelConfig("ownConsRateTotal", "td", 136, SmartHomeUnits.PERCENT));
+        channelConfigsExt
+                .add(new SecondGenerationChannelConfig("autonomyDegreeTotal", "td", 139, SmartHomeUnits.PERCENT));
+
+        channelConfigsExtExt.add(new SecondGenerationChannelConfig("batteryVoltage", "td", 142, SmartHomeUnits.VOLT));
+        channelConfigsExtExt
+                .add(new SecondGenerationChannelConfig("batStateOfCharge", "td", 145, SmartHomeUnits.PERCENT));
+        channelConfigsExtExt
+                .add(new SecondGenerationChannelConfig("selfConsumption", "td", 148, SmartHomeUnits.KILOWATT_HOUR));
+    }
+
+    @SuppressWarnings("null")
+    private void refresh() throws Exception {
+
+        // Create dxsEntries arrays
+        String[] dxsEntries = new String[22];
+        String[] dxsEntriesExt = new String[22];
+        String[] dxsEntriesExtExt = new String[2];
+
+        // Fill dxsEntries with actual values
+        dxsEntries = SecondGenerationDxsEntriesCfg.getDxsEntriesCfg();
+        dxsEntriesExt = SecondGenerationDxsEntriesCfgExt.getDxsEntriesCfgExt();
+        dxsEntriesExtExt = SecondGenerationDxsEntriesCfgExtExt.getDxsEntriesCfgExtExt();
+
+        String dxsEntriesCall = config.url.toString() + "/api/dxs.json?dxsEntries=" + dxsEntries[0];
+        String dxsEntriesCallExt = config.url.toString() + "/api/dxs.json?dxsEntries=" + dxsEntriesExt[0];
+
+        for (int i = 1; i < 23; i++) {
```

Comment:
You shouldn't hardcode `23` here, instead you should use the length of the array you are iterating.